test/random.cc

```diff
@@ -0,0 +1,234 @@
+
+#include "random.hh"
+
+#include <array>
+#include <complex>
+#include "blas.hh"
+#include "blas/util.hh"
+
+
+namespace slate {
+namespace random {
+
```

Comment:
Please add separator rules, i.e.,
```
//------------------------------------------------------------------------------
```
before each function. Note the newer style is no space in `//----`.
About 30% of SLATE's code is still using the old style with space in `// ----`.